src/server/util/spill_file.h

```diff
@@ -0,0 +1,113 @@
+/** Copyright 2020-2021 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef SRC_SERVER_UTIL_SPILL_FILE_H_
+#define SRC_SERVER_UTIL_SPILL_FILE_H_
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include "common/memory/payload.h"
+#include "common/util/status.h"
+#include "common/util/uuid.h"
+#include "io/io/i_io_adaptor.h"
+#include "io/io/io_factory.h"
+#include "io/io/local_io_adaptor.h"
+#include "server/util/file_io_adaptor.h"
+
```

Comment:
I suggest add some docs about the encoding of the spilled file, e.g., how many bytes for headers, etc. in this file.